core_lib/tool/selecttool.cpp

```diff
@@ -5,7 +5,9 @@
 #include "toolmanager.h"
 #include "selecttool.h"
 
-QPointF gSelectionOrigin;
+// Store selection origin so we can calculate
+// the selection rectangle in mousePressEvent.
+static QPointF gSelectionOrigin;
```

Comment:
Why a global variable?
Couldn't you just store the variable as a private in the header or in the function itself.

Global variables should be avoided if possible imo.